vision/vision_quickstart/main.go

```diff
@@ -20,44 +20,153 @@ package main
+	contentTypesToInclude := map[string]bool {
+		"image/CR2": true,
+		"image/gif": true,
+		"image/jpeg": true,
+		"image/png": true,
+		"image/x-icon": true,
+		// "video/3gpp": true,
+		// "video/avi": true,
+		// "video/mp4": true,
+		// "video/mpeg": true,
+		// "video/quicktime": true,
+		// "video/x-ms-wmv": true,
 	}
 
-	labels, err := client.DetectLabels(ctx, image, nil, 10)
-	if err != nil {
-		log.Fatalf("Failed to detect labels: %v", err)
+	// Declare a SafeSearch values to include
+	safeSearchValuesToInclude := map[string]bool {
+		"UNKOWN": true,
+		"POSSIBLE": true,
+		"LIKELY": true,
+		"VERY_LIKELY": true,
+	}
+
+	var names []string
+	it := sourceBucket.Objects(ctx, query)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		if contentTypesToInclude[attrs.ContentType] {
+			names = append(names, attrs.Name)
+		}
 	}
 
-	fmt.Println("Labels:")
-	for _, label := range labels {
-		fmt.Println(label.Description)
+	// Loop through all objects in source bucket and run through SafeSearch
+	// Copy objects that get flagged in SafeSearch
+	for _, name := range names {
+		sourceObjURI := fmt.Sprintf("gs://%s/%s", sourceBucketName, name)
+		fmt.Println(sourceObjURI)
+
+		// Call vision API to detect SafeSearch attributes
+		image := vision.NewImageFromURI(sourceObjURI)
+		props, err := visionClient.DetectSafeSearch(ctx, image, nil)
+		
+		// if err := detectSafeSearchURI(os.Stdout, sourceObjURI); err != nil {
+		// 	fmt.Println("detectSafeSearchURI: %v", err)
+		// }
+
+		if err != nil {
+			fmt.Println("DetectSafeSearch error: %v", err)
+		}
+
+		// props values can be one of:
+		// UNKOWN, VERY_UNLIKELY, UNLIKELY, POSSIBLE, LIKELY, VERY_LIKELY
+		fmt.Fprintln(os.Stdout, "Safe Search properties:")
+		fmt.Fprintln(os.Stdout, "Adult:", props.Adult)
+		fmt.Fprintln(os.Stdout, "Medical:", props.Medical)
+		fmt.Fprintln(os.Stdout, "Racy:", props.Racy)
+		fmt.Fprintln(os.Stdout, "Spoofed:", props.Spoof)
+		fmt.Fprintln(os.Stdout, "Violence:", props.Violence)
+		adult := fmt.Sprintf("%s", props.Adult)
+		racy := fmt.Sprintf("%s", props.Racy)
+		if (safeSearchValuesToInclude[adult]) {
+			if err := copyFile(ioutil.Discard, destinationBucketName, sourceBucketName, name, "adult", adult); err != nil {
+				fmt.Println("copyFile error: %v", err)
+			}
+		}
+		if (safeSearchValuesToInclude[racy]) {
+			if err := copyFile(ioutil.Discard, destinationBucketName, sourceBucketName, name, "racy", racy); err != nil {
+				fmt.Println("copyFile error: %v", err)
+			}
+		}
 	}
 }
 
 // [END vision_quickstart]
+
+// [START storage_copy_file]
+
+// copyFile copies an object into specified bucket.
+func copyFile(w io.Writer, dstBucket, srcBucket, srcObject string, categorization string, confidence string) error {
+	// dstBucket := "bucket-1"
+	// srcBucket := "bucket-2"
+	// srcObject := "object"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
```

Comment:
rather than redefine ctx and client, could pass them thru as args to copyFile.  Most of the Go code i write takes (ctx, c) as the first two arguments, for this reason

Reply:
Ah, good point. I copied this from `https://github.com/GoogleCloudPlatform/golang-samples/blob/main/storage/objects/copy_file.go`. Is there a better way I should be importing that function rather than copy/paste?